packages/spanner/src/transaction.js

```diff
@@ -283,11 +314,28 @@ Transaction.prototype.commit = function(callback) {
  * @param {function} callback - The callback function.
  */
 Transaction.prototype.request = function(config, callback) {
+  var self = this;
+
   var reqOpts = extend({
     session: this.session.formattedName_
   }, config.reqOpts);
 
-  config.method(reqOpts, callback);
+  var gaxOptions = reqOpts.gaxOptions;
+  delete reqOpts.gaxOptions;
+
+  config.method(reqOpts, gaxOptions, function(err, resp) {
```

Comment:
@stephenplusplus I've been thinking about this quite a bit this weekend and I think our best course of action would probably be to treat streams like we do everything else and fail it silently and retry the entire transaction.

At first I thought this was not such a great idea since it could error out mid-stream and our users might have to write code to undo any non-transaction things they might have done, but that pretty much applies to anything done in `runTransaction` at any time.

Because of this, I think we should create a public method that allows the user to get a transaction (`getTransaction`, `beginTransaction`, etc.) that does NOT do all the retry on abort stuff - that way if a user requires a higher level of control they can do whatever it is they want. This all has the added benefit of letting a user use promises.